src/stac_api_validator/validations.py

```diff
@@ -274,6 +276,15 @@ def is_geojson_type(maybe_type: Optional[str]) -> bool:
     )
 
 
+def get_catalog(data_dict: Dict[str, Any], r_session: Session) -> Catalog:
+    stac_io = StacIO.default()
+    if r_session.headers and r_session.headers.get("Authorization"):
+        stac_io.headers = {"Authorization": str(r_session.headers["Authorization"])}
```

Comment:
Why limit this copy to `Authorization`? Any reason not to copy all headers?

Reply:
The session has an `Accept-Encoding` header that pystac doesn't like. An example:
```
{'User-Agent': 'python-requests/2.28.2', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive', 'Authorization': 'api-key <api-key>'}
```
Results in an error:
```
- [Core] Error while traversing Catalog with pystac: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte 
```

If I change the `Accept-Encoding` header to `*` it works fine. I will update it so that this is the case.

Reply:
Oh I missed this response when I was dong my latest review. It still strikes me as strange that you need to set `Accept-Encoding` _only_ when setting authorization. It seems as though we should try to avoid mucking with `Accept-Encoding` if at all possible.

Reply:
Oh it's not only when setting authorization, the same thing would be true even if we were copying over the headers from the session onto pystac's StacIO object, but not setting the `Authorization` header.

It could be a quirk of Planet's STAC API (which besides unit tests, I'm using to test this), let me try with a public API.

Reply:
Could we just

```python
stac_io.headers.update({"Authorization": str(r_session.headers["Authorization"])})
```

so we don't touch the other stuff?

Reply:
Since we made the argument generic (`-H/--headers`) and are now allowing for any custom header to be sent, perhaps we should just copy over the custom ones to pystac?

Reply:
> It could be a quirk of Planet's STAC API (which besides unit tests, I'm using to test this), let me try with a public API

Seems to occur for microsoft planetary computer when that `Accept-Encoding: *` is commented out:
```console
 $ stac-api-validator --root-url https://planetarycomputer.microsoft.com/api/stac/v1 --conformance core --headers "Authorization: api-key fake-api-key" --log-level DEBUG                                                                 
...
Errors:
- [Core] Error while traversing Catalog with pystac: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte 
```

Reply:
@gadomski following up on 
> Since we made the argument generic (-H/--headers) and are now allowing for any custom header to be sent, perhaps we should just copy over the custom ones to pystac?

Should I go ahead with this implementation?

Reply:
Sure!